src/util/macro_logger.rs

```diff
@@ -50,6 +52,88 @@ macro_rules! log_funding_channel_id {
+impl<'a> std::fmt::Display for DebugRoute<'a> {
+	fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+		for (i,h) in self.0.hops.iter().enumerate() {
+			write!(f, "Hop {}\n pubkey {}\n short_channel_id {}\n fee_msat {}\n cltv_expiry_delta {}\n\n", i, log_pubkey!(h.pubkey), h.short_channel_id, h.fee_msat, h.cltv_expiry_delta)?;
+		}
+		Ok(())
+	}
+}
+macro_rules! log_route {
+	($obj: expr) => {
+		::util::macro_logger::DebugRoute(&$obj)
+	}
+}
+
+pub(crate) struct DebugDirectionalChannelInfo<'a>(pub &'a DirectionalChannelInfo);
+impl<'a> std::fmt::Display for DebugDirectionalChannelInfo<'a> {
+	fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+		write!(f, " node id {} last_update {} enabled {} cltv_expiry_delta {} htlc_minimum_msat {} fee_base_msat {} fee_proportional_millionths {}\n", log_pubkey!(self.0.src_node_id), self.0.last_update, self.0.enabled, self.0.cltv_expiry_delta, self.0.htlc_minimum_msat, self.0.fee_base_msat, self.0.fee_proportional_millionths)?;
+		Ok(())
+	}
+}
+macro_rules! log_directionalchannelinfo {
+	($obj: expr) => {
+		::util::macro_logger::DebugDirectionalChannelInfo(&$obj)
+	}
+}
+
+pub(crate) struct DebugChannelInfo<'a>(pub &'a ChannelInfo);
+impl<'a> std::fmt::Display for DebugChannelInfo<'a> {
+	fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+		//TODO: GlobalFeatures
+		write!(f, " one_to_two {}", log_directionalchannelinfo!(self.0.one_to_two))?;
+		write!(f, " two_to_one {}", log_directionalchannelinfo!(self.0.two_to_one))?;
+		Ok(())
+	}
+}
+macro_rules! log_channelinfo {
+	($obj: expr) => {
+		::util::macro_logger::DebugChannelInfo(&$obj)
+	}
+}
+
+pub(crate) struct DebugNodeInfo<'a>(pub &'a NodeInfo);
+impl<'a> std::fmt::Display for DebugNodeInfo<'a> {
+	fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+		write!(f, " Channels\n")?;
+		for c in self.0.channels.iter() {
+			write!(f, " {}\n", c)?;
+		}
+		write!(f, " lowest_inbound_channel_fee_base_msat {}\n", self.0.lowest_inbound_channel_fee_base_msat)?;
+		write!(f, " lowest_inbound_channel_fee_proportional_millionths {}\n", self.0.lowest_inbound_channel_fee_proportional_millionths)?;
+		//TODO: GlobalFeatures, last_update, rgb, alias, addresses
+		Ok(())
+	}
+}
+macro_rules! log_nodeinfo {
+	($obj: expr) => {
+		::util::macro_logger::DebugNodeInfo(&$obj)
+	}
+}
+
+pub(crate) struct DebugNetworkMap<'a>(pub &'a NetworkMap);
+impl<'a> std::fmt::Display for DebugNetworkMap<'a> {
+	fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+		write!(f, "Node id {} network map\n[Channels]\n", log_pubkey!(self.0.our_node_id))?;
+		for (key, val) in self.0.channels.iter() {
+			write!(f, " {} :\n {}\n", key, log_channelinfo!(val))?;
+		}
+		write!(f, "[Nodes]\n")?;
+		for (key, val) in self.0.nodes.iter() {
+			write!(f, " {} :\n {}\n", log_pubkey!(key), log_nodeinfo!(val))?;
+		}
+		Ok(())
+	}
+}
+macro_rules! log_networkmap {
```

Comment:
These are really awesome, but it sucks to add unused code cause it generates way more compile warnings. Honestly its probably worth adding a pub fn Router::log_network function to keep these around.

Reply:
Why not use [allow(dead_code)] and [allow(unused_macros], seems that these attributes been there for avoid lint warnings ?

Reply:
Just adding allow(dead_code) doesn't resolve the issue - the point is that code should be used or it won't be kept up-to-date and probably doesn't belong in the repo at all.